core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -3107,7 +3107,13 @@ Expression getCondition(final List<Expression> argIsNullList) {
       if (nullPolicy == NullPolicy.ARG0) {
         return argIsNullList.get(0);
       }
-      return Expressions.foldOr(argIsNullList);
+      // Condition for regular cases: v0 == null || v1 == null
+      Expression condition = Expressions.foldOr(argIsNullList);
```

Comment:
It can be simplified as
```
if (nullPolicy == NullPolicy.ALL) {
        // Condition for NullPolicy.ALL: v0 == null && v1 == null
        return Expressions.foldAnd(argIsNullList);
      }
// Condition for regular cases: v0 == null || v1 == null
 return Expressions.foldOr(argIsNullList);
```

Reply:
I changed this code block as you suggested. @DonnyZone